src/commonHelper.ts

```diff
@@ -6,57 +6,122 @@
+}
 
 export async function checkProjectVersion(minRequiredSDK: string, maxRequiredSDK: string, projects: string[], commandObserver: CommandObserver) : Promise<string[]> {
-	let unsupportedProjectsMap: string[] = [];
+	let unsupportedProjectsMap = new Array<ProjectInformation>();
     for (let project of projects) {
         let projectFileText = fs.readFileSync(project, 'utf8');
-            let sdkVersion;
-            if ((sdkVersion = regex.exec(projectFileText)) !== null) {
-				if (!(compareVersions.compare(sdkVersion[0], minRequiredSDK, '>=') && compareVersions.compare(sdkVersion[0], maxRequiredSDK, '<='))) {
-                    unsupportedProjectsMap.push(project);
-                }
+        let projectInfo = readXML(project, projectFileText, commandObserver);
+        if (projectInfo) {
+            if (!(compareVersions.compare(projectInfo.sdkVersion, minRequiredSDK, '>=') && compareVersions.compare(projectInfo.sdkVersion, maxRequiredSDK, '<='))) {
+                unsupportedProjectsMap.push(projectInfo);
             }
+        }
     }
 
     if (unsupportedProjectsMap.length > 0) {
 		promptToUpdateVersion(unsupportedProjectsMap, maxRequiredSDK, commandObserver);
 	}
 
-	return Promise.resolve(unsupportedProjectsMap);
+	return Promise.resolve(unsupportedProjectsMap.map(p => p.path));
+}
+
+function readXML(project: string, fileText: string, commandObserver: CommandObserver): ProjectInformation {
+    var parser = new xml2js.Parser();
+    let version: string;
+    let referenceType: sdkReference;
+    parser.parseString(fileText, function(err, result) {
+        if (result && result.Project && result.Project.$ && result.Project.$.Sdk) {
+            referenceType = sdkReference.Attribute;
+            let splitArray: string[] = result.Project.$.Sdk.split('/');
+            if (splitArray.length === 2 && splitArray[0].replace(/\s/g, '').toUpperCase() === sdkName.toUpperCase()) {
+                version = splitArray[1];
+            }
+        }
+        else if(result && result.Project && result.Project.Sdk
+            && result.Project.Sdk[0].$ && result.Project.Sdk[0].$.Name && result.Project.Sdk[0].$.Version) {
+                referenceType = sdkReference.Element;
+                if (result.Project.Sdk[0].$.Name.toUpperCase() === sdkName.toUpperCase()) {
+                    version = result.Project.Sdk[0].$.Version;
+                }
+        }
+    });
+
+    if (version) {
+        return { path: project, sdkVersion: version, sdkReferenceType: referenceType };
+    }
+
+    return undefined;
 }
 
-function promptToUpdateVersion(unsupportedProjects: string[], maxRequiredSDK: string, commandObserver: CommandObserver) {
+function promptToUpdateVersion(unsupportedProjects: Array<ProjectInformation>, maxRequiredSDK: string, commandObserver: CommandObserver) {
     let msg = Constants.unsupportedPostgreSQLSdkMessage;
-    let installItem = 'Update Projects';
+    let installItem = Constants.updateProjectButtonText;
```

Comment:
Localize button text?

Reply:
It is localized in the Constants file

Reply:
Language can probably be changed at run-time and it's better to grab a message from nls every time you need it, rather than storing them in constants.